staking/verify.go

```diff
@@ -41,10 +41,6 @@ var (
 // Ref: https://github.com/golang/go/blob/go1.19.12/src/crypto/x509/x509.go#L793-L797
 // Ref: https://github.com/golang/go/blob/go1.19.12/src/crypto/x509/x509.go#L816-L879
 func CheckSignature(cert *Certificate, msg []byte, signature []byte) error {
-	if err := ValidateCertificate(cert); err != nil {
-		return err
-	}
-
```

Comment:
This is now (maybe) done in certificate parsing.

Reply:
`ValidateCertificate` is still called in `connToIDAndCert` post-durango, no? Or are you referring to something else?

```
func connToIDAndCert(conn *tls.Conn, invalidCerts prometheus.Counter, durangoTime time.Time) (ids.NodeID, net.Conn, *staking.Certificate, error) {
	if err := conn.Handshake(); err != nil {
		return ids.EmptyNodeID, nil, nil, err
	}
	state := conn.ConnectionState()
	if len(state.PeerCertificates) == 0 {
		return ids.EmptyNodeID, nil, nil, errNoCert
	}
	tlsCert := state.PeerCertificates[0]
	// Invariant: ParseCertificate is used rather than CertificateFromX509 to
	// ensure that signature verification can assume the certificate was
	// parseable according the staking package's parser.
	//
	// TODO: Remove pre-Durango parsing after v1.11.x has activated.
	var (
		peerCert *staking.Certificate
		err      error
	)
	if time.Now().Before(durangoTime) {
		peerCert, err = staking.ParseCertificate(tlsCert.Raw)
	} else {
		peerCert, err = staking.ParseCertificatePermissive(tlsCert.Raw)
	}
	if err != nil {
		invalidCerts.Inc()
		return ids.EmptyNodeID, nil, nil, err
	}
	// We validate the certificate here to attempt to make the validity of the
	// peer certificate as clear as possible. Specifically, a node running a
	// prior version using an invalid certificate should not be able to report
	// healthy.
	if err := staking.ValidateCertificate(peerCert); err != nil {
		invalidCerts.Inc()
		return ids.EmptyNodeID, nil, nil, err
	}
	nodeID := ids.NodeIDFromCert(peerCert)
	return nodeID, conn, peerCert, nil
}

Reply:
Good catch. We expect any certificates returned from a `Parse` function to have been validated by that parse function... So I removed the call from `connToIDAndCert`.

Reply:
As for the call in `node` - I'm not exactly sure how we want to handle that in the future yet... We'll probably just re-parse it or something.

Reply:
I think Parse sounds right